src/amo/reducers/reviews.js

```diff
@@ -93,11 +116,68 @@ export const storeReviewObjects = (
   return byId;
 };
 
+type ChangeViewStateParams = {|
+  state: ReviewState,
+  reviewId: number,
+  stateChange: $Shape<ViewStateByReviewId>,
```

Comment:
It's only here to work around this bug: https://github.com/facebook/flow/issues/2405 It's unfortunate that we have to deal with that bug but the project is pretty active so hopefully it will be fixed soon. The Flow developers have chosen not to document `$Shape` (which sucks) but the way they talk about it makes me think it's not going away anytime soon: https://github.com/facebook/flow/issues/875